### DISCUSSION TITLE
Spacy 3.2 floret : the model you're using has no word vectors loaded

### DISCUSSION THREAD

**Discussion by khalidchbab** [explosion/spaCy#10459] [Help: Other Questions] [answered]
Im trying to compare our words vector trained in spacy 3.0 and the new floret vector. The problem is when I load the floret vectors trained in floret mode using the command in spacy 3.2 using the load function and try to calculate similarty for giving words I get an error telling me that the model doesn't contain any vectors. when I checked the meta.json file for the generated vectors I found that the keys in vectors are 0. Is that normal ? if yes, why can't i use the similarity function with the floret model ?

the warning :
_The model you're using has no word vectors loaded, so the result of the Doc.similarity method will be based on the tagger, parser and NER, which may not give useful similarity judgements. This may happen if you're using one of the small models, e.g. `en_core_web_sm`, which don't ship with word vectors and only use context-sensitive tensors. You can always add your own word vectors, or use one of the larger models instead if available._

Example of the meta.json of floret vectors
 `{
  "lang":"fr",
  "name":"pipeline",
  "version":"0.0.0",
  "spacy_version":">=3.2.1,<3.3.0",
  "description":"",
  "author":"",
  "email":"",
  "url":"",
  "license":"",
  "spacy_git_version":"800737b41",
  "vectors":{
    "width":300,
    "vectors":50000,
    "keys":0,
    "name":"fr_pipeline.vectors",
    "mode":"floret"
  },
...
}`

**Comment by adrianeboyd** [comment, marked as answer]
The warning is incorrect and this should be fixed in the next release (v3.3.0) by #10394.

You should still get a similarity score despite the warning, though. If you don't get a similarity score, can you provide a full example where it doesn't work?

**Reply by khalidchbab** [reply]
Thank you for the quick reply. Here is an example of the code im using.
```
nlp1 = spacy.load(model1_path)
nlp1.name = model1_path.split('/')[len(model1_path.split('/'))-1]
print(spacy.info(model1_path))
print(nlp1("Banane").similarity(nlp1("Fruit")))
return 3
```

the program crashes at line 4 and exit with code 1

![Capture](https://user-images.githubusercontent.com/47091620/157261348-c595feca-a3c9-45a8-9386-cedb41532047.PNG)

**Reply by adrianeboyd** [reply]
Please copy and paste text in a code block instead of using screenshots.

This is a little difficult since I can't load the same model to test, but with local models I can't replicate this. It might be another pipeline component that's causing problems?

```python
import spacy

nlp = spacy.load("fi_core_news_md") # this is an early version of an upcoming unpublished v3.3.0 model
print(nlp("Banane").similarity(nlp("Fruit")))
```

And the output looks like this:

```
/tmp/fi.py:4: UserWarning: [W007] The model you're using has no word vectors loaded, so the result of the Doc.similarity method will be based on the tagger, parser and NER, which may not give useful similarity judgements. This may happen if you're using one of the small models, e.g. `en_core_web_sm`, which don't ship with word vectors and only use context-sensitive tensors. You can always add your own word vectors, or use one of the larger models instead if available.
  print(nlp("Banane").similarity(nlp("Fruit")))
0.23333152476930932
```

What happens if you just print the vector for one doc?

```python
print(nlp("Banane").vector)
```

**Reply by khalidchbab** [reply]
Hey, Im sorry to get back to you this late, I tried the code and it worked like expected, I think we have a bug on our end.
``
print(nlp("Banane").similarity(nlp("Fruit")))
``
```
__main__:1: UserWarning: [W007] The model you're using has no word vectors loaded, so the result of the Doc.similarity method will be based on the tagger, parser and NER, which may not give useful similarity judgements. This may happen
 if you're using one of the small models, e.g. en_core_web_sm, which don't ship with word vectors and only use context-sensitive tensors. You can always add your own word vectors, or use one of the larger models instead if available.
0.29646925380652306
```

and when I can get the banane vector

```
[ 3.070559    1.7024813   0.94554     1.0313238  -2.558676    1.0734712 
 -2.092585    1.5642544  -1.2710178  -0.5772118  -0.4795613  -2.7889423 
 -1.7264526  -0.11969197  1.4758569   0.6957609  -0.18328121 -0.14410931
 -0.24046087 -1.0614877  -2.3573768   2.343765   -0.16876064  1.696523  
 -1.8344195  -0.4879394   0.7852325  -1.6384462   0.5868787   1.9879507 
  2.9530718   0.49413437 -0.18304558 -0.26306775 -1.5264719   2.5079443 
 -0.2293163  -1.1225394  -0.03868502 -1.5159231  -0.38992733  0.9038444 
 -0.07203743  0.67608994 -1.4280314   0.3525897  -1.1056956   0.801147
  0.08908439 -2.4247591   1.5961051  -2.022737    0.48067763  0.7706763
 -0.62875074  0.32048246 -0.25729623 -0.28933752 -2.1982799   0.1541281
  1.3651192  -1.6380324   3.4573784  -0.12944117 -1.18434    -0.3904268
  0.54585814  0.3861176  -2.6971455   1.4308887   0.28626877  1.900393
 -1.859442    0.6667706   0.63593143  0.7235457  -0.26293874  2.613924
  2.071345    0.34054366  0.04945939  1.1931204  -2.7233093  -1.8626437
  3.2841105   0.09751251  0.38686568 -1.8340398   0.04370311  0.94799244
 -1.8957671  -0.22266501  0.7100718  -0.13374767 -0.33144486 -1.0162659
 -0.6774481  -1.4753612  -2.1322656   0.8404725  -0.3886444   0.25865507
 -0.21011907 -0.28737146 -0.16255993  0.4991277   0.6047857   0.6396224
 -2.0553606  -0.11416826  0.3253457   1.2200811   2.1366432   1.6495564
  0.46199936 -0.02988246  1.7774142  -1.3651925   0.7154825   0.07384382
 -0.69346094 -0.7525162   0.37707385 -1.7653258  -0.9181658   1.4198813
  0.12508002 -0.42822325 -0.3732925   0.15904048  0.71672    -0.16962184
 -1.847712    1.0966432  -0.05124249 -1.5638894  -0.1142425  -0.8359957
 -0.11439556 -1.4181471  -3.2795346   0.77657175 -0.59184515  1.6485063
 -0.4213128   1.7298548  -0.5484025   2.0894961   1.064187    1.6836905
 -1.5414124   1.2842062  -1.7261717   0.26326066 -1.3815024   0.25354373
  1.129712    0.4337093  -0.30709246 -0.8296787  -1.1530902   0.147345
 -1.0392088  -0.6713356  -0.37404752  0.24839562  1.0602126   0.34106314
  0.8840794   0.3910967  -0.06581935  0.16034561 -1.0740314  -1.6609321
 -1.9662757  -0.73628944 -1.5439074   0.21455249 -2.1056824  -0.82257825
 -0.31441844  0.45916557 -1.4290776   0.34246686  0.46496812  0.6850381 ]
```

thank you for the help
